frontend/catalyst/jax_tracer.py

```diff
@@ -148,9 +148,10 @@ def _make_execution_config(qnode):
     return execution_config
 
 
-def get_device_shots(dev):
-    """Helper function to get device shots."""
-    return dev.shots if isinstance(dev, qml.devices.LegacyDevice) else dev.shots.total_shots
+def get_shots(dev, qnode):
+    """Helper function to get shots."""
+    # pylint: disable=protected-access
+    return qnode._shots if isinstance(dev, qml.devices.LegacyDevice) else qnode._shots.total_shots
```

Comment:
Same comment as above.  I would recommend getting rid of this function entirely.

Reply:
Can't agree more